feature_layers/create-mobile-geodatabase/src/main/java/com/esri/samples/create_mobile_geodatabase/CreateMobileGeodatabaseSample.java

```diff
@@ -0,0 +1,423 @@
+    addFeatureFuture.addDoneListener(() -> {
+      try {
+        addFeatureFuture.get();
+        // update the total feature count on screen if feature was added successfully
+        label.setText("Number of features added: " + featureTable.getTotalFeatureCount());
+        viewTableButton.setDisable(false);
+
+      } catch (InterruptedException | ExecutionException e) {
+        throw new RuntimeException(e);
+      }
+    });
+  }
+
+  /**
+   * Display a new window with the table of features stored in the geodatabase feature table
+   */
+  private void displayTable(Stage stage) {
+
+    // create observable list of type GeoFeature to store the geodatabase features
+    final ObservableList<GeoFeature> fieldData = FXCollections.observableArrayList();
+
+    // query all the features loaded to the table
+    var queryResultFuture = featureTable.queryFeaturesAsync(new QueryParameters());
+    queryResultFuture.addDoneListener(() -> {
+      try {
+        var queryResults = queryResultFuture.get();
+        queryResults.forEach(feature ->
+          // add features to the observable list
+          fieldData.add(new GeoFeature(feature.getAttributes().get("oid").toString(), feature.getAttributes().get("collection_timestamp").toString())));
+      } catch (InterruptedException | ExecutionException e) {
+        throw new RuntimeException(e);
+      }
+
+      // create and set up a new table view to display the features in a table
+      TableView<GeoFeature> table = new TableView<>();
+      table.setEditable(false);
+      table.setVisible(true);
+
+      // create two table columns and add them to the table view
+      TableColumn<GeoFeature, String> oidCol = new TableColumn<>("OID");
+      TableColumn<GeoFeature, String> timeCol = new TableColumn<>("COLLECTION TIMESTAMP");
+      table.getColumns().add(oidCol);
+      table.getColumns().add(timeCol);
+
+      // associate data to the table columns referencing the fields in the GeoFeature class
+      oidCol.setCellValueFactory(new PropertyValueFactory<>("oid"));
+      timeCol.setCellValueFactory(new PropertyValueFactory<>("timestamp"));
+
+      // add data to the table view
+      table.setItems(fieldData);
+
+      // create a StackPane, Scene, and Stage for displaying the table view in a new window
+      var pane = new StackPane();
+      pane.getChildren().add(table);
+      var scene = new Scene(pane, 220, 230);
+
+      // set up stage properties before display
+      tableStage = new Stage();
+      tableStage.setTitle("Features");
+      tableStage.setScene(scene);
+      tableStage.setX(stage.getX() + 200);
+      tableStage.setY(stage.getY() + 100);
+      tableStage.show();
+      showTableWindow = true; // to control window behaviour
+    });
+  }
+
+  /**
+   * Stops and releases all resources used in application.
+   */
+  @Override
+  public void stop() {
+
+    if (mapView != null) {
+      mapView.dispose();
```

Comment:
let's also close the geodatabase here and delete the geodatabase from the users machine on exit. Alternatively consider using the temporary directories approach to let the system handle file tidying up as we do in our other samples: this would be the preferred scenario for consistency but the approach you have here works: worth checking if this works from the sample viewer too. 

```suggestion
      mapView.dispose();
      geodatabase.close();
            try {
        Files.deleteIfExists(geodatabasePath);
      } catch (IOException e) {
        e.printStackTrace();
      }
```